server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -752,28 +806,43 @@ def generate_token(
             del batch
             raise e
 
+        try:
+            out, speculative_logits = out.logits, out.speculative_logits
+        except Exception:
+            out = out
+            speculative_logits = None
```

Comment:
Maybe this can be refactored or at least we can add a comment/have a more precise except than the blanket `Exception`.